pythonFiles/vscode_datascience_helpers/getVariableInfo/vscodeGetVariableInfo.py

```diff
@@ -58,3 +58,16 @@ def _VSCODE_getVariableProperties(var, listOfAttributes):
         if hasattr(var, attr)
     }
     return _VSCODE_json.dumps(result)
+
+
+def _VSCODE_getVariableTypes(vars, varnames):
```

Comment:
How long does this take to execute for say 10000 variables?

Reply:
I was wondering if we're making another request for types, could we do the list and the types at the same time?

Reply:
Internally this would essentially call the same function we do now to list the names (you'd have to look up what the magic %_who_ls does in ipykernel)

Reply:
think I got something that works well